test/Chainweb/Test/Pact/RemotePactTest.hs

```diff
@@ -0,0 +1,272 @@
+{-# LANGUAGE MultiWayIf #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+
+-- |
+-- Module: Chainweb.Test.RemotePactTest
+-- Copyright: Copyright © 2019 Kadena LLC.
+-- License: See LICENSE file
+-- Maintainer: Mark Nichols <mark@kadena.io>
+-- Stability: experimental
+--
+-- Unit test for Pact execution via the Http Pact interface (/send, etc.)(inprocess) API  in Chainweb
+module Chainweb.Test.Pact.RemotePactTest where
+
+import Control.Concurrent hiding (readMVar, putMVar)
+import Control.Concurrent.Async
+import Control.Concurrent.MVar.Strict
+import Control.Exception
+import Control.Lens
+import Control.Monad
+
+import qualified Data.Aeson as A
+import qualified Data.ByteString as BS
+import qualified Data.HashMap.Strict as HM
+import Data.Maybe
+import Data.Proxy
+import Data.Streaming.Network (HostPreference)
+import Data.String.Conv (toS)
+
+import Network.HTTP.Client.TLS as HTTP
+import Network.Connection as HTTP
+
+import Numeric.Natural
+
+import Servant.Client
+import System.FilePath
+import System.LogLevel
+import System.Time.Extra
+
+import Test.Tasty.HUnit
+import Test.Tasty
+import Test.Tasty.Golden
+
+import Text.RawString.QQ(r)
+
+import Pact.Types.API
+
+-- internal modules
+
+import Chainweb.Chainweb
+import Chainweb.Chainweb.PeerResources
+import Chainweb.Graph
+import Chainweb.HostAddress
+import Chainweb.Logger
+import Chainweb.Miner.Config
+import Chainweb.NodeId
+import Chainweb.Pact.RestAPI
+import Chainweb.Payload.PayloadStore (emptyInMemoryPayloadDb)
+import Chainweb.Test.P2P.Peer.BootstrapConfig
+import Chainweb.Test.Pact.Utils
+import Chainweb.Utils
+import Chainweb.Version
+
+import P2P.Node.Configuration
+import P2P.Peer
+
+apiSend :: SubmitBatch -> ClientM RequestKeys
+apiSend = client (Proxy :: Proxy SendApi)
+
+apiPoll :: Poll -> ClientM PollResponses
+apiPoll = client (Proxy :: Proxy PollApi)
+
+nNodes :: Natural
+nNodes = 1
+
+tests :: IO TestTree
+tests = do
+    peerInfoVar <- newEmptyMVar
+    let cwVersion = TestWithTime petersonChainGraph
+    theAsync <- async $ runTestNodes Warn cwVersion nNodes Nothing peerInfoVar
+    link theAsync
+    newPeerInfo <- readMVar peerInfoVar
+    let thePort = _hostAddressPort (_peerAddr newPeerInfo)
+
+    putStrLn $ "Server listening on port: " ++ show thePort ++ ". Sending client request..."
```

Comment:
Let's avoid chatty tests. 

In other places we use HUnit progress settings as backend for the Chainweb-logger. In multi node tests there is some CPP setup to log to the console when compiled with `-dDEBUG`. So, those are options. 

But tests run in parallel and if each tests prints to the console it's not helpful always.